docs/en/programming-model.md

```diff
@@ -201,6 +201,78 @@ try {
+  3. Select Field: a `string` field, rendered as a drop down, that only accepts an option defined in the associated `enum` key; useful if orgs can only select a configuration from a number of pre-defined options
+
+**Important**: Configuration values should **never contain sensitive information**. This feature is **only for configuration** and as such, does not provide sufficient security to support storage of secrets and other sensitive information.
+
+All three options share the following keys:
+  * `key`: the **unique** key your app will use to refer to a particular configuration value
+  * `label`: the label text that users will see when configuring a particular setting
+  * `description`: text describing the behavior that a specific setting will modify; displayed to users when configuring your app
+  * `default`: the default configuration value for a setting if the user has **not** already input one
+    * In the case of a select field, the `default` **must** be one of the options associated with the `enum` key
+  * `required`: whether the field is required for the app to function; if set to `true`, users will **not** be able to use your app until they've configured the required setting
+    * While a setting with both a non-empty `default` value and `required: true` is valid, requiring the setting is unnecessary since the `default` value satisfies the `required: true` constraint even when the user has never configured the setting
+
+For a more concrete illustration, assuming that your app's custom configuration is defined as follows:
+
+```JSON
+[
+    {
+        "key": "boolean_field",
+        "label": "A Boolean Field",
+        "description": "A boolean field that controls app behavior",
+        "required": false,
+        "type": "boolean"
+    },
+    {
+        "key": "input_field",
+        "label": "An Input Field",
+        "description": "An input field that controls app behavior",
+        "required": false,
+        "type": "string"
+    },
+    {
+        "key": "select_field",
+        "label": "A Select Field",
+        "description": "A select field that controls app behavior",
+        "enum": [
+            "choice_a",
+            "choice_b"
+        ],
+        "required": false,
+        "type": "string"
+    }
+]
+```
+
+An org using your app would see the following page to configure the settings you defined:
+
+<img style="max-width:80%" alt="Configuring an App" src="https://user-images.githubusercontent.com/17037651/161990413-7f5061b0-6577-4dc3-b775-cf6d14e864d8.png">
+
+Back in your app, you could access configured values with the following:
+
+```js
+const customConfig = await client.config.getConfig()
+/* customConfig = {
+  boolean_field: <boolean>,
+  input_field: <string>,
+  select_field: <string>
+} */
```

Comment:
Maybe show an example of getting the value of a specific field?

Reply:
Works; I'll update that in a bit